Python/optimizer_bytecodes.c

```diff
@@ -846,6 +846,15 @@ dummy_func(void) {
        next = sym_new_type(ctx, &PyLong_Type);
     }
 
+    op(_CALL_TYPE_1, (callable, null, arg -- res)) {
+        if (sym_has_type(arg)) {
+            res = sym_new_const(ctx, (PyObject *)sym_get_type(arg));
+        }
```

Comment:
Is there an easy way to test this part? i.e. that the `res` becomes const? AFAIK there are no other uops/guards that check for instances of `type` so we can't test it indirectly by checking for removed uops..

Reply:
Yeah, none check for instances of `type` specifically... mostly because we haven't separated these out or added optimizer cases for them yet! ;)

I think we could still use the fact that it's a const to our advantage, though. Haven't tried this, but you might be able to write a test that does `t = type(foo); if t is not None: ...`. I'm pretty sure the JIT will remove the `_GUARD_IS_NOT_NONE_POP` instruction when the value is a known constant like this.